.github/workflows/amzn1.yml

```diff
@@ -0,0 +1,105 @@
+name: Update BTFHub Archive for Amazon Linux 1
+on:
+  schedule:
+    - cron: "0 1 * * *"
+  workflow_dispatch: {}
+jobs:
+  amazon-update:
+    name: update amazon linux 1 btfhub-archive
+    runs-on: ubuntu-latest
+    container:
+      image: amazonlinux:1
+    steps:
+      - name: Install needed amazon packages
+        run: |
+          yum install -y yum-utils clang wget bzip2 zlib-devel m4 xz gzip cmake find git
+          yum-config-manager -y --enable amzn-main-debuginfo amzn-updates-debuginfo
+          yum-config-manager --add-repo https://cli.github.com/packages/rpm/gh-cli.repo
+          yum install -y gh
+        shell: bash
+
+      - name: Authenticate
+        run: |
+          gh auth login --with-token <<<'${{ secrets.DD_BTFHUB_BOT_GITHUB_TOKEN }}'
+
+      - name: Check out BTFHub
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.DD_BTFHUB_BOT_GITHUB_TOKEN }}
+          submodules: 'recursive'
+
+      - run: git config --system --add safe.directory $(pwd)
+        shell: bash
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: build elfutils
+        run: |
+          wget -nv https://sourceware.org/elfutils/ftp/0.190/elfutils-0.190.tar.bz2
+          tar xf elfutils-0.190.tar.bz2
+          cd elfutils-0.190
+          ./configure --disable-libdebuginfod --disable-debuginfod --disable-demangler
+          make install
+          ldconfig
+
+      - name: build tar
+        run: |
+          wget -nv https://ftpmirror.gnu.org/tar/tar-1.35.tar.xz
+          tar xf tar-1.35.tar.xz
+          cd tar-1.35
+          FORCE_UNSAFE_CONFIGURE=1 ./configure --prefix=/
+          make install
+
+      - name: install newer clang
```

Comment:
Why do we need clang?

Reply:
There are some `cgo` parts in btfhub that need it.